django/core/management/commands/listurls.py

```diff
@@ -0,0 +1,275 @@
+import json
+from importlib import import_module
+from io import StringIO
+
+from django.conf import settings
+from django.contrib.admindocs.views import (
+    extract_views_from_urlpatterns,
+    simplify_regex,
+)
+from django.core.management import color
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+from django.utils import termcolors
+
+FORMATS = (
+    "aligned",
+    "table",
+    "verbose",
+    "json",
+    "pretty-json",
+)
+
+
+COLORLESS_FORMATS = (
+    "json",
+    "pretty-json",
+)
+
+
+class Command(BaseCommand):
+    help = "List URL patterns in the project with optional filtering by prefixes."
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.style.HEADER = termcolors.make_style(opts=("bold",))
+        self.style.ROUTE = termcolors.make_style(opts=("bold",))
+        self.style.VIEW_NAME = termcolors.make_style(fg="yellow", opts=("bold",))
+        self.style.NAME = termcolors.make_style(fg="red", opts=("bold",))
+
+    def add_arguments(self, parser: CommandParser):
+        super().add_arguments(parser)
+
+        # Sorting
```

Comment:
Typically "what" comments are discourage 👍

Reply:
I'm not sure what counts as "typical," but in my experience, they provide great value as visual anchors and help decrease the mental load.

Reply:
I'll tell you what if you find your commenting style _prolifically present in the way you do it in this PR_ anywhere in the Django codebase (there are the occasional what comments) I'll gladly eat my words 🤣 but it's highly likely that you'll be asked to remove them hence "typically". I don't have the final say that's up to the Django fellows...

In my experience what comments are just visual noise that end up diverging and become misleading or just outright incorrect.

But don't take my word for it: https://blog.codinghorror.com/code-tells-you-how-comments-tell-you-why/

Reply:
@shangxiao Thanks for the details. I'll remove them if the consensus is not to have them, but I don't think that what that blog post describes applies here. Repeating the code in plan English is different from summarizing a code block by a short title. I was going for the second case.

Reply:
Thanks for the opinion.

Reply:
> I'll remove them if the consensus is not to have them

Please do not mark comments as resolved if there has been no agreement as it is difficult for other reviewers to reply.

I agree these comments should be removed as the argument names and help text are descriptive enough and we do not add such comments for usages of `parser.add_argument`

Reply:
I removed the comments from the `parser.add_argument()` blocks. I also tried removing the other seemingly redundant comments, but I think the code is better without them. Let me know if there are any other comments you think should be removed.